python/cuml/cuml/cluster/hdbscan/hdbscan.pyx

```diff
@@ -620,34 +622,129 @@ class HDBSCAN(UniversalBase, ClusterMixin, CMajorInputTagMixin):
+            "gen_min_span_tree": model.gen_min_span_tree,
+            "prediction_data": model.prediction_data,
+        }
+
+    def _params_to_cpu(self):
+        return {
+            "min_cluster_size": self.min_cluster_size,
+            "min_samples": self.min_samples,
+            "cluster_selection_epsilon": self.cluster_selection_epsilon,
+            "max_cluster_size": self.max_cluster_size,
+            "metric": self.metric,
+            "alpha": self.alpha,
+            "p": self.p,
+            "cluster_selection_method": self.cluster_selection_method,
+            "allow_single_cluster": self.allow_single_cluster,
+            "gen_min_span_tree": self.gen_min_span_tree,
+            "prediction_data": self.prediction_data,
+        }
+
+    def _attrs_from_cpu(self, model):
+        if (raw_data_cpu := getattr(model, "_raw_data", None)) is None:
+            # Fit with precomputed metric
+            raise UnsupportedOnGPU
+
+        if not isinstance(raw_data_cpu, np.ndarray):
+            # Sparse input
+            raise UnsupportedOnGPU
+
+        raw_data = to_gpu(raw_data_cpu, order="C", dtype="float32")
+        labels = to_gpu(model.labels_, order="C", dtype="int32")
+        state = _HDBSCANState.from_sklearn(self.handle, model, raw_data)
+        if model._prediction_data is not None:
+            state.generate_prediction_data(self.handle, raw_data, labels)
+
+        return {
+            "labels_": labels,
+            "probabilities_": to_gpu(model.probabilities_),
+            "cluster_persistence_": to_gpu(model.cluster_persistence_),
+            "_raw_data": raw_data,
+            "_raw_data_cpu": raw_data_cpu,
+            "_state": state,
+            "n_clusters_": state.n_clusters,
+            "_single_linkage_tree": model._single_linkage_tree,
+            "_min_spanning_tree": model._min_spanning_tree,
+            "_prediction_data": model._prediction_data,
+            **super()._attrs_from_cpu(model),
+        }
+
+    def _attrs_to_cpu(self, model):
+        out = {
+            "labels_": to_cpu(self.labels_),
+            "probabilities_": to_cpu(self.probabilities_),
+            "cluster_persistence_": to_cpu(self.cluster_persistence_),
+            "_condensed_tree": self._condensed_tree,
+            "_single_linkage_tree": self._single_linkage_tree,
+            "_min_spanning_tree": self._min_spanning_tree,
+            "_raw_data": self._get_raw_data_cpu(),
+            "_all_finite": True,
+            **super()._attrs_to_cpu(model),
+        }
+        if self.prediction_data:
+            out["_prediction_data"] = self.prediction_data_
+        return out
+
     def __init__(self, *,
                  min_cluster_size=5,
                  min_samples=None,
                  cluster_selection_epsilon=0.0,
                  max_cluster_size=0,
                  metric='euclidean',
                  alpha=1.0,
-                 p=2,
+                 p=None,
```

Comment:
Why the change? or really, why was it not `None` which is the default in `hdbscan.HDBSCAN`? bug or on purpose?

Reply:
We actually never reference `p` in cuml, it's a fully ignored parameter (since we don't support other metrics than euclidean). Changing the defaults to match seemed fine and made the interop tests easier.